src/error.rs

```diff
@@ -9,68 +9,85 @@ use thiserror::Error;
 #[derive(Error, Debug)]
 #[non_exhaustive]
 pub enum GeoArrowError {
+    /// Incorrect type was passed to an operation.
     #[error("Incorrect type passed to operation: {0}")]
     IncorrectType(Cow<'static, str>),
 
     /// Returned when functionality is not yet available.
     #[error("Not yet implemented: {0}")]
     NotYetImplemented(String),
 
+    /// General error.
```

Comment:
I haven't been particularly good at selecting errors and so I think the vast majority of errors in this crate are "General" errors 😅

Reply:
IMO that's just fine ... feels like a natural step on the way from **anyhow** -> **thiserror**.